script/upgrade/GoerliM2Upgrade.s.sol

```diff
@@ -0,0 +1,121 @@
+import "../../src/contracts/pods/EigenPodManager.sol";
+import "../../src/contracts/pods/DelayedWithdrawalRouter.sol";
+
+import "../../src/contracts/permissions/PauserRegistry.sol";
+import "../../src/contracts/middleware/BLSPublicKeyCompendium.sol";
+
+import "../../src/test/mocks/EmptyContract.sol";
+import "../../src/test/mocks/ETHDepositMock.sol";
+
+import "forge-std/Script.sol";
+import "forge-std/Test.sol";
+
+// # To load the variables in the .env file
+// source .env
+
+// # To deploy and verify our contract
+// forge script script/upgrade/GoerliM2Deployment.s.sol:GoerliM2Deployment --rpc-url $RPC_URL  --private-key $PRIVATE_KEY --broadcast -vvvv
+contract GoerliM2Deployment is Script, Test {
+    Vm cheats = Vm(HEVM_ADDRESS);
+
+    string public deploymentOutputPath = string(bytes("script/output/M1_deployment_goerli_2023_3_23.json"));
+
+    address executorMultisig;
+    address operationsMultisig;
+    address pauserMultisig;
+    address beaconChainOracleGoerli = 0x40B10ddD29a2cfF33DBC420AE5bbDa0649049f2c;
+
+    IETHPOSDeposit public ethPOS;
+
+    ISlasher public slasher;
+    IDelegationManager public delegation;
+    DelegationManager public delegationImplementation;
+    IStrategyManager public strategyManager;
+    StrategyManager public strategyManagerImplementation;
+    IEigenPodManager public eigenPodManager;
+    EigenPodManager public eigenPodManagerImplementation;
+    IDelayedWithdrawalRouter public delayedWithdrawalRouter;
+    IBeacon public eigenPodBeacon;
+    EigenPod public eigenPodImplementation;
+
+    function run() external {
+        // read and log the chainID
+        uint256 chainId = block.chainid;
+        emit log_named_uint("You are deploying on ChainID", chainId);
+
+        // READ JSON DEPLOYMENT DATA
+        string memory deployment_data = vm.readFile(deploymentOutputPath);
+        slasher = Slasher(stdJson.readAddress(deployment_data, ".addresses.slasher"));
+        delegation = slasher.delegation();
+        strategyManager = slasher.strategyManager();
+        eigenPodManager = strategyManager.eigenPodManager();
+        delayedWithdrawalRouter = DelayedWithdrawalRouter(stdJson.readAddress(deployment_data, ".addresses.delayedWithdrawalRouter"));
+        eigenPodBeacon = eigenPodManager.eigenPodBeacon();
+        ethPOS = eigenPodManager.ethPOS();
+
+
+        vm.startBroadcast();
+        delegationImplementation = new DelegationManager(strategyManager, slasher, eigenPodManager);
+        strategyManagerImplementation = new StrategyManager(delegation, eigenPodManager, slasher);
+        eigenPodManagerImplementation = new EigenPodManager(
+            ethPOS,
+            eigenPodBeacon,
+            strategyManager,
+            slasher,
+            delegation
+        );
+        eigenPodImplementation = new EigenPod(
+            ethPOS,
+            delayedWithdrawalRouter,
+            eigenPodManager,
+            31 gwei,
+            0.5 gwei
+        );
```

Comment:
might want the more explicit syntax here, to extra-ensure that the last two args can't be in the wrong order

Reply:
Addressed [d9a548c](https://github.com/Layr-Labs/eigenlayer-contracts/pull/177/commits/d9a548c6fab99936f3c71a0aac68bbdee21bcce7)